provider/scripts/k8s/offline_store_pandas_runner.py

```diff
@@ -318,98 +386,212 @@ def get_code_from_file(mode, file_path, etcd_credentials):
     return code
 
 
-def get_blob_credentials(args):
-    """
-    Retrieve credentials from the blob store.
-    Parameters:
-        args: Namespace (input arguments that passed via environment variables)
-
-    Output:
-        credentials: Namespace(type="", ...) (includes credentials needed for each blob store.)
-    """
-
-    if args.mode == K8S_MODE and args.azure_blob_credentials:
-        return Namespace(
-            type=AZURE,
-            connection_string=args.azure_blob_credentials,
-            container=args.azure_container_name,
-        )
-    else:
-        return Namespace(
-            type=LOCAL,
-        )
-
-
 def get_etcd_host(host, ports):
     """
     Converts the host and ports list into the expected for need for etcd client.
     Parameters:
         host: str   ("127.0.0.1")
         port: [str] (["2379"])
-    Output:
-        etcd_host: [(str, str)] ([("127.0.0.1", "2379")])
+
+    Returns:
+        etcd_host: [(str, str)] (example: [("127.0.0.1", "2379")])
     """
     etcd_host = []
     for port in ports:
         etcd_host.append((host, int(port)))
     return tuple(etcd_host)
 
 
+def get_blob_store(store_credentials):
+    """
+    Returns a BlobStore object based on the store_credentials type
+    Parameters:
+        store_credentials: Namespace (used to download/upload files)
+
+    Returns:
+        BlobStore
+    """
+
+    if store_credentials.type == S3:
+        return S3BlobStore(store_credentials)
+    elif store_credentials.type == AZURE:
+        return AzureBlobStore(store_credentials)
+    elif store_credentials.type == LOCAL:
+        return LocalBlobStore(store_credentials)
+    else:
+        raise Exception(f"blob store type {store_credentials.type} is not supported.")

```

Comment:
Possibly nit-picky, but I wonder if it would be preferrable to create a `StoreType` enum class that has a constructor method to support this. Might avoid adding more `elif`s in the future and could be reused elsewhere if needed.

Reply:
Based on our discussion, I no longer feel this point makes sense at this time. Thanks for the context.